packages/components/src/color-picker/index.js

```diff
@@ -106,140 +113,137 @@ const dataToColors = ( oldColors, { source, valueKey, value } ) => {
 		source,
 		...{ ...oldColors[ source ], ...{ [ valueKey ]: value } },
 	};
-};
-
-export default class ColorPicker extends Component {
-	constructor( { color = '0071a1' } ) {
-		super( ...arguments );
-		const colors = colorToState( color );
-		this.state = {
-			...colors,
-			draftHex: toLowerCase( colors.hex ),
-			draftRgb: colors.rgb,
-			draftHsl: colors.hsl,
-		};
-		this.commitValues = this.commitValues.bind( this );
-		this.setDraftValues = this.setDraftValues.bind( this );
-		this.resetDraftValues = this.resetDraftValues.bind( this );
-		this.handleInputChange = this.handleInputChange.bind( this );
-	}
-
-	commitValues( data ) {
-		const { oldHue, onChangeComplete = noop } = this.props;
+}
 
+export default function ColorPicker( {
+	color = '0071a1',
+	className,
+	disableAlpha,
+	oldHue,
+	onChangeComplete = noop,
+} ) {
+	const initialColor = colorToState( color );
+	const [ colors, setColors ] = useState( {
+		...initialColor,
+		draftHex: toLowerCase( initialColor.hex ),
+		draftRgb: initialColor.rgb,
+		draftHsl: initialColor.hsl,
+	} );
+
+	const debouncedOnChangeComplete = useCallback(
+		useDebounce( onChangeComplete, 100 ),
+		[ onChangeComplete ]
+	);
+
+	const { hex, hsl, hsv, rgb, draftHex, draftHsl, draftRgb } = colors;
+
+	const classes = classnames( className, {
+		'components-color-picker': true,
+		'is-alpha-disabled': disableAlpha,
+		'is-alpha-enabled': ! disableAlpha,
+	} );
+
+	function commitValues( data ) {
 		if ( isValidColor( data ) ) {
-			const colors = colorToState( data, data.h || oldHue );
-			this.setState(
-				{
-					...colors,
-					draftHex: toLowerCase( colors.hex ),
-					draftHsl: colors.hsl,
-					draftRgb: colors.rgb,
-				},
-				debounce( partial( onChangeComplete, colors ), 100 )
-			);
+			const newColors = colorToState( data, data.h || oldHue );
+			setColors( {
+				...newColors,
+				draftHex: toLowerCase( newColors.hex ),
+				draftHsl: newColors.hsl,
+				draftRgb: newColors.rgb,
+			} );
+
+			debouncedOnChangeComplete( newColors );
```

Comment:
```suggestion
```
The current implementation passes this as a callback for `.setState` so it's asynchronous. To match that in the refactor this could be called via an effect hook (`useUpdateEffect` from '../utils' to avoid the initial render). This would simplify creating a stable `commitValues` as in my preceding comment and is the main reason I'm suggesting it.